account_spread_cost_revenue/models/account_invoice_line.py

```diff
@@ -87,9 +87,11 @@ def _filter_line(aline, iline):
             spread_type = (
                 'sale' if line.invoice_type in ['out_invoice', 'out_refund']
                 else 'purchase')
-            spread_auto = self.env['account.spread.template.auto'].search(
-                [('template_id.auto_spread', '=', True),
-                 ('template_id.spread_type', '=', spread_type)])
+            spread_auto = self.env['account.spread.template.auto'].sudo().search([
```

Comment:
If the sudo is necessary here, isn't it also necessary when `template_id` is accessed later in `account.spread.invoice.line.link.wizard`'s confirm method?

Reply:
My original idea was to avoid failures when not any auto spread templates are present for these invoice lines.
If there are not auto spread template, there's no reason to raise the access rights error and so in this case we have to make sure that the search on the auto spread templates model should be possible, regardless of the user's rights.

Instead, the case of confirming the `account.spread.invoice.line.link.wizard`, we are creating a spread table. This is something that may requires explicit access rights, depending on the customization we want. I would keep the current access right for the wizard confirm method.

Reply:
OK, clear now. Can you add a short comment
```
# Search with sudo() so that no template access is required in order
# to confirm invoices to which no template applies
```